logging.go

```diff
@@ -41,7 +41,7 @@ func DefaultLogger(clioCfg Config, store redact.Store) (logger.Logger, error) {
 
 	l, err := logrus.New(
 		logrus.Config{
-			EnableConsole: cfg.Verbosity > 0 && !cfg.Quiet,
```

Comment:
Is the problem that `cfg.Verbosity` is never set, so defaults to `0` and therefore `EnableConsole == false`? Is there a way that this configuration could be consolidated to just be based on a verbosity number instead of 3 distinct settings of verbosity, quiet, and level?

Reply:
> Is the problem that cfg.Verbosity is never set, so defaults to 0 and therefore EnableConsole == false?

Yes, that's the problem. So if there's no TTY _and_ there's no verbosity, there's nowhere for `log.Warn` outputs to go, because `EnableConsole: false` will cause the underlying writer to be `io.Discard`.

> Is there a way that this configuration could be consolidated to just be based on a verbosity number instead of 3 distinct settings of verbosity, quiet, and level?

I don't think it can be simplified beyond this. We want `-q` but a log file and a verbosity set in the environment to still write to the environment I think? I'm not sure I understand your suggestion.

Reply:
I'm not suggesting to change any of the exposed options but rather make the options determine one specific verbosity level, and base all logging configuration on that. It looks like we have a function to do that: https://github.com/anchore/clio/blob/main/logging.go#L133 -- this takes all 3 options into consideration and returns a single logging level. We could modify this function to use the result of that function as the logging level, where anywhere we use quiet, instead we use `levelFromSelectLevel != logger.Disabled` or something like that, and we use that level to set `Level:` below, etc.

Reply:
This TODO actually seems to be saying we _don't_ want to collapse the amount of options here. 

> make the options determine one specific verbosity level, and base all logging configuration on that

I don't think we can sensibly do that, because we have 4 independent dimensions of input:

1. `quiet` - don't do fancy terminal UI or log to the terminal
2. `-v` / `-vv` raise the level of the logger, regardless of where it prints
3. Whether stdout/stderr are a TTY - determines whether we can do fancy TUI stuff, but shouldn't determine anything else
4. `logfile` - change the logger, whatever its level is, to append to this path instead of stderr, regardless of whether `--quiet` was passed or a TTY is present.

In other words, something like `SYFT_LOG_FILE=/tmp/log.txt syft --quiet -vvv ...` should be a valid command, that should append trace logs of `/tmp/log.txt` and not do any fancy TUI stuff or log to stdout or stderr.

Does that make sense? Maybe we should have a discussion about how these options should interact, then revisit this PR?

Reply:
[Here](https://anchorecommunity.discourse.group/t/request-for-discussion-logging-levels-and-when-to-use-them/240?u=willmurphy) is the longer discussion about logging.

Reply:
Actually [this](https://anchorecommunity.discourse.group/t/rfd-configuring-output/241?u=willmurphy) is the topic I meant to link to.